usaspending_api/search/v2/views/spending_by_category_views/spending_by_agency_types.py

```diff
@@ -32,16 +36,23 @@ def build_elasticsearch_result(self, response: dict) -> List[dict]:
         agency_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
         for bucket in agency_info_buckets:
             agency_info = json.loads(bucket.get("key"))
-
-            results.append(
-                {
-                    "amount": int(bucket.get("sum_field", {"value": 0})["value"]) / Decimal("100"),
-                    "name": agency_info.get("name"),
-                    "code": agency_info.get("abbreviation"),
-                    "id": agency_info.get("id"),
-                }
-            )
-
+            result = {
+                "amount": int(bucket.get("sum_field", {"value": 0})["value"]) / Decimal("100"),
+                "name": agency_info.get("name"),
+                "code": agency_info.get("abbreviation"),
+                "id": agency_info.get("id"),
+            }
+            if self.agency_type == AgencyType.AWARDING_TOPTIER:
```

Comment:
Can you leave a comment over this logic similar to what you have in your PR description? 
```
Only returns a non-null value if the agency has a profile page - meaning it is an agency that has at least one submission.
```